### PR TITLE
assets: harden tapd client lifecycle and RFQ handling

### PR DESCRIPTION
**Pull request by hieblmi**
## Summary

  Improve the tapd client’s lifecycle, timeout handling, concurrency, and validation.

  - Close tapd gRPC connections during daemon shutdown, initialization failures, and `view` completion.
  - Validate RFQ timeout durations before creating the client.
  - Round positive fractional timeouts up to tapd’s whole-second representation.
  - Allow cached asset-name reads while another universe lookup is in progress.
  - Reject nil, malformed, non-positive, and invalid-scale RFQ asset rates instead of risking a panic.
  - Add regression tests for timeout boundaries, concurrent cache access, and malformed rates.

### CODE REVIEW

File: loopd/daemon.go [hunk 1]
```diff
@@ -166,13 +166,23 @@ func (d *Daemon) Start() error {
...
 		// likely LiT is already running and blocking the DB
 		return fmt.Errorf("%v: make sure no other loop daemon process "+
 			"(standalone or embedded in lightning-terminal) is"+
 			"running", err)
 	}
 	if err != nil {
+		if d.assetClient != nil {
+			d.assetClient.Close()
+			d.assetClient = nil
+		}
```
Comment (starius) [inline loopd/daemon.go]: I propose to add `defer` inside `if d.cfg.Tapd.Activate` block above and close it there.

File: assets/client.go [hunk 2]
```diff
@@ -288,6 +294,26 @@ func getPaymentMaxAmount(satAmount btcutil.Amount, feeLimitMultiplier float64) (
 	)
 }
 
+// getRfqTimeoutSeconds converts the configured RFQ timeout to the whole
+// seconds accepted by tapd. Sub-second precision is rounded up so a positive
+// timeout can never become tapd's invalid zero value.
```
Comment (starius) [inline assets/client.go]: Isn't there a stronger reason to do it so we don't get a timeout error?

If we account on 1.5s timeout (and actively using it, e.g. use it at time 1.4s, i.e. 0.1s before the timeout), but tapd enforces the timeout of just 1s (rounded down, as before), we get a timeout error. So we round up to request enough time, right?

File: assets/client.go [hunk 3]
```diff
@@ -254,6 +264,19 @@ func (c *TapdClient) GetAssetPrice(ctx context.Context, assetID string,
...
+	}
+
+	coefficient, ok := new(big.Int).SetString(assetRate.Coefficient, 10)
+	if !ok {
+		return 0, fmt.Errorf("invalid asset rate coefficient: %q",
+			assetRate.Coefficient)
+	}
+	if coefficient.Sign() <= 0 {
+		return 0, fmt.Errorf("asset rate coefficient must be positive")
+	}
```
Comment (starius) [inline assets/client.go:316]: We need a similar hardening for GetRfqForAsset. Here is an example how to fix it: https://gist.github.com/starius/c23dd24330c3d36c9e852d2f4596313f

File: assets/client.go [hunk 4]
```diff
@@ -163,12 +170,13 @@ func (c *TapdClient) GetRfqForAsset(ctx context.Context,
...
 
-	c.assetNameMutex.Lock()
-	defer c.assetNameMutex.Unlock()
 	assetIdStr := hex.EncodeToString(assetId)
+	c.assetNameMutex.RLock()
 	if name, ok := c.assetNameCache[assetIdStr]; ok {
+		c.assetNameMutex.RUnlock()
 		return name, nil
 	}
+	c.assetNameMutex.RUnlock()
```
Comment (starius) [inline assets/client.go:172]: Let's preserve the `defer` pattern somehow. Maybe make private accessor methods for this map? To encapsulate map+mutex work there.

**Review by starius — APPROVED** [review]
**LGTM!** 💾

Left come comments. The main one is hardening for GetRfqForAsset.

File: loopd/daemon.go (hunk 1 above)
Comment (hieblmi) [reply loopd/daemon.go]: done

File: assets/client.go (hunk 2 above)
Comment (hieblmi) [reply assets/client.go]: done

File: assets/client.go (hunk 3 above)
Comment (hieblmi) [reply assets/client.go:316]: extracted the checks and re-used it for GetFrqForAsset.

File: assets/client.go (hunk 4 above)
Comment (hieblmi) [reply assets/client.go:172]: done
